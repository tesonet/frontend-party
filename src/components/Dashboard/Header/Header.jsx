import React, { Component } from "react";

import logo from "assets/images/testio.svg";
import styles from "./Header.module";

export default class DashboardHeader extends Component {
  handleLogout() {
    localStorage.removeItem("testioToken");
    this.props.props.history.push("/");
  }
  render() {
    return (
      <header className={styles.header}>
        <figure className={styles.header__logo}>
          <img src={logo} alt="Testio" width="115" height="30" />
        </figure>
        <div className={styles.header__action}>
          <button
            onClick={this.handleLogout.bind(this)}
            className={styles.header__action__logout}
          >
            Logout
          </button>
        </div>
      </header>
    );
  }
}
