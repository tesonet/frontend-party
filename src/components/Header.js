import React, { Component } from 'react';
import withStyles from '@material-ui/core/styles/withStyles';
import PropTypes from 'prop-types';
import AppLogoB from '../images/logo-b.svg';

// MUI Stuff
import Grid from '@material-ui/core/Grid';
import Button from '@material-ui/core/Button';
import Tooltip from '@material-ui/core/Tooltip';

//Icons
import ExitToAppIcon from '@material-ui/icons/ExitToApp';

// Redux Stuff
import { connect } from 'react-redux';
import { logoutUser } from '../redux/actions/userActions';

const styles = theme => ({
  ...theme.spreadThis
});

class Header extends Component {
  handleLogout = () => {
    this.props.logoutUser();
  };

  render() {
    const { classes } = this.props;
    return (
      <div className='header'>
        <Grid
          direction='row'
          container
          justify='space-between'
          alignItems='center'
          spacing={0}
          className={classes.headerTop}
        >
          <Grid item sm>
            <img src={AppLogoB} alt='appImg' className='logoImage' />
          </Grid>
          <Grid item sm className={classes.logoutButton}>
            <Tooltip title='Logout' placement='top'>
              <Button
                onClick={this.handleLogout}
                className={classes.logoutBtn}
                startIcon={<ExitToAppIcon />}
              >
                Logout
              </Button>
            </Tooltip>
          </Grid>
        </Grid>
      </div>
    );
  }
}
const mapActionsToProps = { logoutUser };

Header.propTypes = {
  classes: PropTypes.object.isRequired,
  logoutUser: PropTypes.func.isRequired
};

export default connect(null, mapActionsToProps)(withStyles(styles)(Header));
