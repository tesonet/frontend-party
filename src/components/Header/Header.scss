nav {
    height: 100px;
}

.logout-link {
    cursor: pointer;

    img {
        margin-right: 10px;
    }
}
