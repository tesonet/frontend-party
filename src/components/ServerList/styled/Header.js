import styled from 'styled-components';

const Header = styled.header`
  align-items: center;
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  min-height: 7rem;
`;

export default Header;
