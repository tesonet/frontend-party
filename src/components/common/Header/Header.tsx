import React from 'react';
import { RouteComponentProps } from 'react-router-dom';
import styles from './Header.module.scss';
import logo from './assets/logo-testio-small.png';
import logOut from './assets/logout.svg';
import { withRouter } from 'react-router-dom';
import { ROUTES } from '../../../routes';

interface Props extends RouteComponentProps<any> {}
const HeaderComponent: React.FC<Props> = ({ history }) => {
  const onLogOut = () => {
    localStorage.removeItem('auth-token');
    history.push(ROUTES.login);
  };
  return (
    <div className={styles.container}>
      <img data-test="header-logo" src={logo} alt="Testio" />
      <button
        data-test="logout-button"
        className={styles.logoutButton}
        onClick={onLogOut}
      >
        <img src={logOut} alt="logout" />
        Logout
      </button>
    </div>
  );
};

export const Header = withRouter(HeaderComponent);
