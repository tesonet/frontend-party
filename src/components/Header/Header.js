import React from 'react';
import PropTypes from 'prop-types';
import Logo from '../UI/Logo';
import IconButton from '../UI/IconButton';
import './Header.scss';

export const Header = ({ onButtonClick }) => (
  <header className="header">
    <Logo type="servers" />
    <IconButton buttonText="Logout" icon="logout" onClick={onButtonClick} />
  </header>
);

Header.propTypes = {
  onButtonClick: PropTypes.func.isRequired,
};

export default Header;
