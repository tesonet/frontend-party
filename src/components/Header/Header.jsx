import React from "react";
import { PropTypes } from "prop-types";
import { Navbar, NavbarBrand } from "reactstrap";
import logo from "../../assets/images/logo-dark.svg";
import icon from "../../assets/icons/ico-lock.svg";
import "./Header.scss";

const Header = ({ onLogout }) => (
    <Navbar>
        <NavbarBrand>
            <img src={logo} className="header-logo" />
        </NavbarBrand>
        <a className="logout-link" onClick={onLogout}>
            <img src={icon} />
            Logout
        </a>
    </Navbar>
);

export default Header;

Header.propTypes = {
    onLogout: PropTypes.func
};
