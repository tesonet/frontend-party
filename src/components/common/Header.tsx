import React from 'react';
import Box from '../styled/Box';
import styled from 'styled-components';
import { ReactComponent as LogoSVG } from '../../assets/logotype-testio_black.svg';

const HeaderStyled = styled(Box)`
  height: 113px;
  display: flex;
  align-items: center;
`;

const Logo = styled(LogoSVG)`
  display: block;
  height: 30px;
  margin-left: 15px;
`;

export default function Header() {
  return (
    <HeaderStyled>
      <Logo alt="Tesonet" />
    </HeaderStyled>
  );
}
