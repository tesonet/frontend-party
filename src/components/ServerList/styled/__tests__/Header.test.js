import React from 'react';
import { shallow } from 'enzyme';

import Header from '../Header';

describe('Header component', () => {
  it('should render the <header /> element.', () => {
    const renderedComponent = shallow(<Header />);

    expect(renderedComponent.type()).toEqual('header');
    expect(renderedComponent).toMatchSnapshot();
  });
});
