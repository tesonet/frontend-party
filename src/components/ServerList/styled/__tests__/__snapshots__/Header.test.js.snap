// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component should render the <header /> element. 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <styled.header />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "className": "sc-bdVaJa joFqqu",
    },
    "ref": null,
    "rendered": null,
    "type": "header",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "className": "sc-bdVaJa joFqqu",
      },
      "ref": null,
      "rendered": null,
      "type": "header",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
